session/session.go

```diff
@@ -976,35 +976,37 @@ func (s *session) GetGlobalSysVar(name string) (string, error) {
 	return sysVar, nil
 }
 
-// SetGlobalSysVar implements GlobalVarAccessor.SetGlobalSysVar interface.
-func (s *session) SetGlobalSysVar(name, value string) error {
-	if name == variable.TiDBSlowLogMasking {
-		name = variable.TiDBRedactLog
+// SetGlobalSysVarOnly updates the internal value without calling the SetGlobal hook.
+// This is useful if you want to prevent recursion in the case that you want to keep two variables in sync.
+func (s *session) SetGlobalSysVarOnly(name, value string) (err error) {
```

Comment:
Just comment. It seems weird for me that `SetGlobalSysVarOnly` is called inside `SetGlobal`... And `SetGlobalSysVarOnly` is not an internal method but a public one.

Maybe we can automatic the setting of session/global vars after all the cleanup. It is still annoying if you need to `.systems[xx] = xx` and `SetGlobalSysVarOnly(xx, xx)` for every variable.

Reply:
Using an `Alias` field was my original plan, but when I thought about it some more, there are some cases that are not well handled by this. For example, MySQL has a variable called `log-warnings` which is deprecated for `log-error-verbosity` (which makes more sense), but the value of `log-warnings=2` maps to `log-error-verbosity=3`. We could however use an `Alias`/`SynonymFor:` field for the simple cases, and leave the complicated as `SetGlobalSysVarOnly` etc.

For deprecation, the reason I didn't add a field, is that the variable might be deprecated entirely, deprecated for another variable, or just a specific value is deprecated. Currently there is a helper function that handles one of the major cases (deprecated for)

Reply:
I have changed this to use `Aliases` (plural). It still needs the `SetGlobalSysVarOnly` function, but the `SetGlobalSysVar` no longer calls it directly. They both call an internal function which is shared for the update.